### PR TITLE
Add nbqa-ruff for doc notebooks

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/pymc-labs/pymc-marketing/issues/554

I suggest we use `nbqa-ruff` (https://github.com/nbQA-dev/nbQA) in the pre-commit hooks. I left out clv/dev notebooks as they have many ruff-errors to fix, and we can do it on a separate issue (BTW, what do we want to do with these dev notebooks ?)

----
📚 Documentation preview 📚: https://pymc-marketing--561.org.readthedocs.build/en/561/

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -15,6 +15,13 @@ repos:
         args: [--ignore-missing-imports]
         files: ^pymc_marketing/
         additional_dependencies: [numpy>=1.20, pandas-stubs]
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.7.1
+    hooks:
+    - id: nbqa-ruff
+      args: ["--fix", "--show-source"]
+      files: ^docs/source/notebooks/
+      exclude: ^docs/source/notebooks/clv/dev/
```
Comment (wd60622) [inline .pre-commit-config.yaml:24]: These aren't on the documentation site anyways, right?

**Review by wd60622 — APPROVED** [review]
Awesome. Thanks for taking this @juanitorduz

**Comment by juanitorduz** [comment]
No, they are experimental 😄

File: .pre-commit-config.yaml (hunk 1 above)
Comment (juanitorduz) [reply .pre-commit-config.yaml:24]: No, the are not

**Pull request merged**